proposals/utils/pdf_diff_logic.py

```diff
@@ -0,0 +1,1271 @@
+                            self.new_object.obj.proposal,
+                        )
+                    )
+                elif field in study_list:
+                    rows.append(
+                        DiffRow(
+                            field, self.old_object.obj.study, self.new_object.obj.study
+                        )
+                    )
+                else:
+                    rows.append(
+                        DiffRow(field, self.old_object.obj, self.new_object.obj)
+                    )
+            return rows
+
+
+class ExtraDocumentsSection(BaseSection):
+    """This class receives an Documents object.
+    Overrides the __init__ to create a formatted section title"""
+
+    row_fields = [
+        "informed_consent",
+        "briefing",
+        "director_consent_declaration",
+        "director_consent_information",
+        "parents_information",
+    ]
+
+    def __init__(self, obj, num):
+        super().__init__(obj)
+        self.section_title = _("Extra formulieren ") + str(num + 1)
+
+    def get_row_fields(self):
+        rows = copy(self.row_fields)
+        obj = self.obj
+
+        if not obj.informed_consent:
+            rows.remove("informed_consent")
+        if not obj.briefing:
+            rows.remove("briefing")
+        if not obj.director_consent_declaration:
+            rows.remove("director_consent_declaration")
+        if not obj.director_consent_information:
+            rows.remove("director_consent_information")
+        if not obj.parents_information:
+            rows.remove("parents_information")
+
+        return rows
+
+
+class DMPFileSection(BaseSection):
+    """This class receives a proposal object
+    Also unnecessary I suppose. But I though why not ..."""
+
+    section_title = _("Data Management Plan")
+
+    row_fields = ["dmp_file"]
+
+
+class EmbargoSection(BaseSection):
+    """Gets passed a proposal object"""
+
+    section_title = _("Aanmelding versturen")
+
+    row_fields = ["embargo", "embargo_end_date"]
+
+    def get_row_fields(self):
+        rows = copy(self.row_fields)
+        obj = self.obj
+
+        if not obj.embargo:
+            rows.remove("embargo_end_date")
+
+        return rows
+
+
+class CommentsSection(BaseSection):
+    """Gets passed a proposal object. 
+    """
+
+    section_title = _("Ruimte voor eventuele opmerkingen")
+
+    row_fields = ["comments"]
+
+
+def get_extra_documents(obj):
+    """A function to retrieve all extra documents for a specific proposal."""
+    from studies.models import Documents
+
+    extra_documents = []
+
+    for document in Documents.objects.filter(proposal=obj, study__isnull=True):
+        extra_documents.append(document)
+
+    return extra_documents
+
+
+def create_context_pdf(context, model):
+    """A function to create the context for the PDF, which gets called in the ProposalAsPdf view."""
+
+    sections = []
+
+    sections.append(GeneralSection(model))
+    sections.append(WMOSection(model.wmo))
```

Comment:
(Same holds true for the diff variant)

This fails on pre-approved studies, which do not get a WMO object.